pkg/apis/policy/v1alpha1/types_certificaterequestpolicy.go

```diff
@@ -178,24 +178,33 @@ type CertificateRequestPolicyAllowedX509Subject struct {
 }
 
 // CertificateRequestPolicyAllowedStringSlice represents an allowed string
-// slice value paired with whether the field is a required value on the
-// request.
+// slice value and/or field CEL validations paired with whether the field is a
+// required value on the request.
 type CertificateRequestPolicyAllowedStringSlice struct {
 	// Defines the values that are permissible to be present on request.
 	// Accepts wildcards "*".
 	// An omitted field or value of `nil` forbids any value on the related field
-	// in the request from being requested.
-	// An empty slice `[]` is equivalent to `nil`, however an empty slice pared
-	// with Required `true` is an impossible condition that always denies.
-	// Values may not be `nil` if Required is `true`.
+	// in the request from being requested unless Validations are present.
+	// Values and Validations may not be both be empty if Required is `true`.
 	// +optional
 	Values *[]string `json:"values,omitempty"`
```

Comment:
I don't know why this field is defined as a **pointer** of string slice. IMO treating nil and empty slice differently in an API is an anti-pattern, but I presume there is a reason for this here? 🤔

Reply:
It would be very useful to properly document the current behaviour. Imo the current documentation is not very easy to understand.

Reply:
> It would be very useful to properly document the current behaviour.

I agree with this - even if the behavior is well documented on our website. @inteon, are you suggesting a pre-PR to brush up on the current Go docs? IMO this should be written with the Kubernetes API in mind. As it stands now, it's more a documentation of the Go API.

Reply:
If you could create such a pre-PR this, that would be awesome!

Reply:
https://github.com/cert-manager/approver-policy/pull/280